In [1]:
import pandas as pd
# Load the data from the excel file
df = pd.read_excel('data.xlsx')


In [16]:
# Display the different columns in the dataset
print(df.columns)
print()
print(df.head())
print(df.shape)

missing_values = df.isnull().sum()
print("Missing values in each column:")
print(missing_values)




Index(['Education Level', 'Specialization', 'Skills', 'Certifications',
       'CGPA/Percentage', 'Recommended Career'],
      dtype='str')

  Education Level    Specialization                                   Skills  \
0      Bachelor's           Finance  Counseling, MS Office, Machine Learning   
1    Intermediate           Science                    Accounting, MS Office   
2        Master's          Business           Accounting, SQL, Data Analysis   
3      Bachelor's  Computer Science                            Communication   
4          Matric          Business                            Data Analysis   

         Certifications  CGPA/Percentage Recommended Career  
0             Tally ERP               67   Business Analyst  
1         AWS Certified               67  Software Engineer  
2  Mental Health Basics               90  Financial Analyst  
3                   NaN               75              Clerk  
4             Tally ERP               83    Sales Assistant  
(5000,

The dataset has 5000 records with 6 columns. It's currently only missing values in certifications, which represents that the person does not have any certifications.

In [31]:
print(df.nunique())
print()
print(df["Education Level"].unique())
print()
print(df["Specialization"].unique())
print()
print(df["Certifications"].unique())
print()

print(df["Recommended Career"].unique())

Education Level         5
Specialization          8
Skills                757
Certifications          7
CGPA/Percentage        36
Recommended Career     12
dtype: int64

<StringArray>
['Bachelor's', 'Intermediate', 'Master's', 'Matric', 'PhD']
Length: 5, dtype: str

<StringArray>
[         'Finance',          'Science',         'Business',
 'Computer Science',             'Arts',       'Psychology',
         'Commerce',      'Engineering']
Length: 8, dtype: str

<StringArray>
[            'Tally ERP',         'AWS Certified',  'Mental Health Basics',
                     nan,     'Digital Marketing',           'CFA Level 1',
      'Creative Writing', 'Google Data Analytics']
Length: 8, dtype: str

<StringArray>
[   'Business Analyst',   'Software Engineer',   'Financial Analyst',
               'Clerk',     'Sales Assistant',           'Professor',
 'Marketing Executive',   'Junior Accountant',    'School Counselor',
 'Data Entry Operator',  'Research Scientist',         'ML Engineer']

In [ ]:
unique_skills = (
    df["Skills"]
      .str.split(",")
      .explode()
      .str.strip()
      .unique()
)

print(len(unique_skills))
print(unique_skills)

10
<StringArray>
[        'Counseling',          'MS Office',   'Machine Learning',
         'Accounting',                'SQL',      'Data Analysis',
      'Communication', 'Financial Analysis',             'Python',
          'Marketing']
Length: 10, dtype: str


All columns are standardized , ready to be used. Skills has 10 unique values, the reason it showed 757 unique values is because its counting different combinations of the 10 values as unique.

## 5 number summary on data to detect any outliers or anomolies in data and perform actions to remove them

## Exploratory Data analysis,
for example: most common careers, amount of each degree type, most common skills etc